src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -400,10 +403,231 @@ struct npc_instructor_zhi : public ScriptedAI
     TaskScheduler _scheduler;
 };
 
+enum JaominRoMisc
+{
+    // Spells
+    SPELL_CSA_AREATRIGGER_DUMMY      = 88811,
+    SPELL_HAWK_DIVING_TO_EARTH       = 108955,
+    SPELL_HAWK_DIVING_TO_EARTH_DMG   = 108935,
+    SPELL_BABY_ELEPHANT_TAKES_A_BATH = 108938,
+    SPELL_FORCE_SUMMONER_TO_RIDE     = 108583,
+    SPELL_EJECT_ALL_PASSENGERS       = 50630,
+    SPELL_DIZZY                      = 108959,
+    SPELL_RIDE_DRAKE                 = 108582,
+    SPELL_SERVERSIDE_KILL_CREDIT     = 109837,
+    SPELL_FULL_HEALTH                = 17683,
+
+    // Texts
+    SAY_INTRO                        = 0,
+    SAY_DEFEATED                     = 1,
+
+    // Movement
+    POINT_RANDOM_DEST                = 0,
+
+    // Events
+    EVENT_RANDOM_SPELL               = 1,
+    EVENT_MOVE,
+    EVENT_HEAL,
+    EVENT_MOVE_HOME
+};
+
+// 54611 - Jaomin Ro
+struct npc_jaomin_ro : public ScriptedAI
+{
+    npc_jaomin_ro(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_RANDOM_SPELL, 4s, 6s);
+    }
+
+    void JustReachedHome() override
+    {
+        if (me->HasUnitFlag(UNIT_FLAG_NON_ATTACKABLE_2))
+            me->RemoveUnitFlag(UNIT_FLAG_NON_ATTACKABLE_2);
```

Comment:
Don't both with HasFlag before RemoveFlag if there is no other code behind this condition